third_party/gpus/cuda_configure.bzl

```diff
@@ -526,6 +526,27 @@ def _find_cuda_lib(lib, repository_ctx, cpu_value, basedir, version="",
   auto_configure_fail("Cannot find cuda library %s" % file_name)
 
 
+def _find_cupti_header_dir(repository_ctx, cuda_config):
+  """Returns the path to the directory containing cupti.h
+
+  On most systems, the cupti library is not installed in the same directory as
+  the other CUDA libraries but rather in a special extras/CUPTI directory.
+
+  Args:
+    repository_ctx: The repository context.
+    cuda_config: The CUDA config as returned by _get_cuda_config
+
+  Returns:
+    The path of the directory containing the cupti header.
+  """
+  cuda_toolkit_path = cuda_config.cuda_toolkit_path
```

Comment:
Let's make it easier to add new alternate paths like this here:

```
cupti_alternate_paths = [ ....... ] # We can even define this list as a constant at the top of this file.

for path in cupti_alternate_paths:
  if repository_ctx.path(cuda_toolkit_path + path).exists:
    return ....

auto_configure_fail....
```

Same comment for all the modified methods.

Reply:
Done.

Thank you for the suggestion, the file looks a lot saner now!